app/helpers/checkout_helper.rb

```diff
@@ -70,9 +70,11 @@ def display_checkout_tax_total(order)
   def display_checkout_taxes_hash(order)
     totals = OrderTaxAdjustmentsFetcher.new(order).totals
 
-    totals.each_with_object({}) do |(tax_rate, tax_amount), hash|
-      hash[number_to_percentage(tax_rate.amount * 100, precision: 1)] =
-        Spree::Money.new tax_amount, currency: order.currency
+    totals.map do |tax_rate, tax_amount|
+      {
+        amount: Spree::Money.new(tax_amount, currency: order.currency),
+        percentage: number_to_percentage(tax_rate.amount * 100, precision: 1),
+      }
     end
```

Comment:
Newb question: the idea with converting the array to a hash is to be able to call `amount` or `percentage` separately, from the same object, right?

Reply:
Yes. A hash has unique keys. So if we use the percentage as hash key we can store only one of them. But a list can just have as many items as we need. And the view turns it into a list of rows in the HTML. So the data structure is much closer to what's needed to render.